mllib/src/main/scala/org/apache/spark/mllib/feature/ChiSqSelector.scala

```diff
@@ -57,22 +69,21 @@ class ChiSqSelectorModel @Since("1.3.0") (
    * Preserves the order of filtered features the same as their indices are stored.
    * Might be moved to Vector as .slice
    * @param features vector
-   * @param filterIndices indices of features to filter
+   * @param filterIndices indices of features to filter, must be ordered asc
    */
   private def compress(features: Vector, filterIndices: Array[Int]): Vector = {
-    val orderedIndices = filterIndices.sorted
```

Comment:
I think there's a better solution still -- accept any input, sort it and store the sorted array. I do want to avoid this unnecessary assertion about the input but yeah this can be optimized. I should have pushed on that in the initial review since it was my intent, and yeah it matters actually.